src/mlir/verona-mlir.cc

```diff
@@ -97,7 +97,8 @@ namespace
 
     // Default grammar
     // FIXME: Move to llvm::sys::path, but LLVM's GetMainExecutable is horrible
-    grammarFile = path::directory(path::executable()).append("/grammar.peg");
+    if (grammarFile.empty())
```

Comment:
Do we ever change the grammar for tests?  If not, I am not sure why we need the tests to supply it?  It seems you are adding functionality here, where as we could just drop the argument from the tests?

Reply:
This change is irrelevant here and makes no difference in the CMake files or tests. Leave this for another PR.

Reply:
I disagree. The update-dump script was broken because it was passing the `-g` flag to `verona-mlir`. I can fix this by removing it from update-dump or by adding the flag to the source, but either approach seems “relevant” to me.

Reply:
I would prefer to remove from `update-dump`.

Reply:
@mjp42 I was mostly trying to make the way tests are run uniform across different binaries. `verona-ast` is called with the grammar file, so I figured so should `verona-mlir`. I’m happy to just remove it from the cmake script instead.

To be honest I’m not really sure why we allow the path to the grammar file to be configurable (and in my opinion it should even be embedded in the binary).

Reply:
The comment says: "Grammar file is not optional" and you're changing it to optional. This is change in behaviour that does not belong to a CMake change.

If the update script is using the program in the wrong way, we need to fix the update script, not the program.

If we want to change the behaviour to make grammar files optional, that's another discussion.

Reply:
I really feel like this is splitting hairs. Saying commits that touch cmake scripts aren’t allowed to change implementation is such an arbitrary rule.

Additionally this line of code is in a function called `cmdLineDefaults`, along with a comment saying this is the “default grammar”. That makes it sound like it was intended to be a flag all along, but was just forgotten. “The update script is using the program in the wrong way” is nothing but the other side of the “the program doesn’t accept options it is supposed to” coin.

Whatever. I’ll change the cmake script and undo the C++ change.